### PR TITLE
Issue #2894259 by WidgetsBurritos: Validate Sitemap URL on Submit

### PR DESCRIPTION
**Pull request by WidgetsBurritos**
Drupal Issue: https://www.drupal.org/node/2894259

This PR validates urls and sitemaps on submission.

### CODE REVIEW

**Comment by bighappyface** [comment]
@WidgetsBurritos please see the status check failure related to code style

File: src/Form/WebPageArchiveEditForm.php [hunk 1]
```diff
@@ -31,11 +32,13 @@ class WebPageArchiveEditForm extends WebPageArchiveFormBase {
...
    * @param \Drupal\Core\Entity\EntityStorageInterface $web_page_archive_storage
    *   The storage.
+   * @param \Drupal\web_page_archive\Parser\SitemapParser $sitemap_parser
+   *   The sitemap parser service.
    * @param \Drupal\web_page_archive\Plugin\CaptureUtilityManager $capture_utility_manager
    *   The capture utility manager service.
    */
-  public function __construct(EntityStorageInterface $web_page_archive_storage, CaptureUtilityManager $capture_utility_manager) {
-    parent::__construct($web_page_archive_storage);
+  public function __construct(EntityStorageInterface $web_page_archive_storage, SitemapParser $sitemap_parser, CaptureUtilityManager $capture_utility_manager) {
```
Comment (bighappyface) [inline src/Form/WebPageArchiveEditForm.php]: How about making the `SitemapParser` the third parameter with a null default? Sitemap usage is optional and making it a required param within method parameters seems heavy. The same would apply to `WebPageArchiveFormBase` but the parameter is already at the end, making it easy to convert to an optional parameter.

Comment (WidgetsBurritos) [reply src/Form/WebPageArchiveEditForm.php]: I can do so, but it is worth noting that the `create()` method below this always injects that service in anyway. So I'm not entirely sure we're buying much in terms of performance here?

Comment (bighappyface) [reply src/Form/WebPageArchiveEditForm.php]: Not concerned about performance, just method signatures. Adding parameters into the middle tends to reflect an API break, and form classes extending this (although there really aren't many) would be impacted. Fair?

Comment (WidgetsBurritos) [reply src/Form/WebPageArchiveEditForm.php]: Fair enough.  Pushing

**Comment by WidgetsBurritos** [comment]
@bighappyface I addressed your feedback in a separate commit (this will get squashed when we merge the code): 

f6204d4ac71d6dc840b43cb4b1c59e9e9caeabdf

**Review by bighappyface — APPROVED** [review]
+1 for 988dd89
+1 for f6204d4

**Pull request merged**